docs/source/examples/cim.rst

```diff
@@ -0,0 +1,260 @@
+algorithm, experience pool, and a set of parameters that governs the training loop. For this scenario, the agent is the
+abstraction of a port. We choose DQN as our underlying learning algorithm with a TD-error-based sampling mechanism.
+
+.. code-block:: python
+    class CIMAgent(AbsAgent):
+        ...
+        def train(self):
+            if len(self._experience_pool) < self._min_experiences_to_train:
+                return
+
+            for _ in range(self._num_batches):
+                indexes, sample = self._experience_pool.sample_by_key("loss", self._batch_size)
+                state = np.asarray(sample["state"])
+                action = np.asarray(sample["action"])
+                reward = np.asarray(sample["reward"])
+                next_state = np.asarray(sample["next_state"])
+                loss = self._algorithm.train(state, action, reward, next_state)
+                self._experience_pool.update(indexes, {"loss": loss})
+
+Agent Manager
+-------------
+
+`Agent manager <https://maro.readthedocs.io/en/latest/key_components/rl_toolkit.html#agent-manager>`_ is an agent
+assembler and isolates the complexities of the environment and algorithm. For this scenario, It will load the DQN
+algorithm and an experience pool for each agent.
+
+.. code-block:: python
+
+    class DQNAgentManager(AbsAgentManager):
+        def _assemble(self, agent_dict):
+            set_seeds(config.agents.seed)
+            num_actions = config.agents.algorithm.num_actions
+            for agent_id in self._agent_id_list:
+                eval_model = LearningModel(decision_layers=MLPDecisionLayers(name=f'{agent_id}.policy',
+                                                                             input_dim=self._state_shaper.dim,
+                                                                             output_dim=num_actions,
+                                                                             **config.agents.algorithm.model)
+                                           )
+
+                algorithm = DQN(model_dict={"eval": eval_model},
+                                optimizer_opt=(RMSprop, config.agents.algorithm.optimizer),
+                                loss_func_dict={"eval": smooth_l1_loss},
+                                hyper_params=DQNHyperParams(**config.agents.algorithm.hyper_parameters,
+                                                            num_actions=num_actions))
+
+                experience_pool = ColumnBasedStore(**config.agents.experience_pool)
+                agent_dict[agent_id] = CIMAgent(name=agent_id, algorithm=algorithm, experience_pool=experience_pool,
+                                                **config.agents.training_loop_parameters)
+
+Main Loop with Actor and Learner (Single Process)
+-------------------------------------------------
+
+This single-process workflow of a learning policy's interaction with a MARO environment is comprised of:

```

Comment:
tailing with Noun phrase? How about

- Initialization of ...
- Definition of ..
- Creation of ..

Reply:
Using initializing, defining, creating ... instead